packages/malloy-render/src/html/table.ts

```diff
@@ -190,12 +234,40 @@ export class HTMLTableRenderer extends ContainerRenderer {
             );
             cells[rowIndex][columnIndex] = childRenderer.createHeaderCell(
               nonDimension.field,
-              shouldTranspose
+              shouldTranspose,
+              {
+                name: nonDimension.flattenedField?.name,
+                childRenderer: nonDimension.flattenedField?.getChildRenderer(
+                  childRenderer.childRenderers
+                ),
+              }
             );
             columnIndex++;
           }
         }
         pivotDepth = Math.max(pivotDepth, dimensions!.length);
+      } else if (shouldFlatten) {
```

Comment:
will you always be able to flatten here? is there any condition you'd like to check to see if flattening is a valid option?

Reply:
Done